### PR TITLE
ptable: make linux-32g;linux-64g maximum system

### PR DESCRIPTION
**Pull request by mengzhuo**

### CODE REVIEW

File: l-loader/generate_ptable.sh [hunk 1]
```diff
@@ -214,12 +214,8 @@ case ${PTABLE} in
...
         #[9: trustfirmware: 369M-371M]
         fakeroot ${SGDISK} -n 9:0:+2M -t 9:0700 -u 9:f1e126a6-ceef-45c1-aace-29f33ac9cf13 -c 9:"trustfirmware" ${TEMP_FILE}
-        #[10: system: 371M-5059M]
-        fakeroot ${SGDISK} -n 10:0:+4688M -t 10:8300 -u 10:c3e50923-fb85-4153-b925-759614d4dfcd -c 10:"system" ${TEMP_FILE}
-        #[11: vendor: 5059M-5843M]
-        fakeroot ${SGDISK} -n 11:0:+784M -t 11:0700 -u 11:919d7080-d71a-4ae1-9227-e4585210c837 -c 11:"vendor" ${TEMP_FILE}
-        #[12: reserved: 5843M-5844M]
-        fakeroot ${SGDISK} -n 12:0:+1M -t 12:0700 -u 12:611eac6b-bc42-4d72-90ac-418569c8e9b8 -c 12:"reserved" ${TEMP_FILE}
+        #[10: system: 371M-END]
+        fakeroot ${SGDISK} -n -E -t 10:8300 -u 10:c3e50923-fb85-4153-b925-759614d4dfcd -c 10:"system" ${TEMP_FILE}
```
Comment (hzhuang1) [inline l-loader/generate_ptable.sh:218]: Why do you want to drop vendor and reserved partition?

File: l-loader/generate_ptable.sh [hunk 2]
```diff
@@ -232,12 +228,8 @@ case ${PTABLE} in
...
         #[11: trustfirmware: 417M-419M]
         fakeroot ${SGDISK} -n 11:0:+2M -t 11:0700 -u 11:f1e126a6-ceef-45c1-aace-29f33ac9cf13 -c 11:"trustfirmware" ${TEMP_FILE}
-        #[12: system: 419M-5107M]
-        fakeroot ${SGDISK} -n 12:0:+4688M -t 12:8300 -u 12:c3e50923-fb85-4153-b925-759614d4dfcd -c 12:"system" ${TEMP_FILE}
-        #[13: vendor: 5107M-5891M]
-        fakeroot ${SGDISK} -n 13:0:+784M -t 13:0700 -u 13:919d7080-d71a-4ae1-9227-e4585210c837 -c 13:"vendor" ${TEMP_FILE}
-        #[14: reserved: 5891M-5892M]
-        fakeroot ${SGDISK} -n 14:0:+1M -t 14:0700 -u 14:611eac6b-bc42-4d72-90ac-418569c8e9b8 -c 14:"reserved" ${TEMP_FILE}
+        #[12: system: 419M-END]
+        fakeroot ${SGDISK} -n -E -t 12:8300 -u 12:c3e50923-fb85-4153-b925-759614d4dfcd -c 12:"system" ${TEMP_FILE}
```
Comment (hzhuang1) [inline l-loader/generate_ptable.sh:232]: Why do you want to drop vendor and reserved partition?

Comment (mengzhuo) [reply l-loader/generate_ptable.sh:232]: I simply don't need them and it works fine for my board

```
Linux hikey970 4.9.78-147538-g244928755bbe #1 SMP PREEMPT Thu May 31 10:58:02 CST 2018 aarch64 GNU/Linux
```

```
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        59G  3.1G   53G   6% /
devtmpfs        2.8G     0  2.8G   0% /dev
tmpfs           2.9G     0  2.9G   0% /dev/shm
tmpfs           2.9G   17M  2.8G   1% /run
tmpfs           5.0M  4.0K  5.0M   1% /run/lock
tmpfs           2.9G     0  2.9G   0% /sys/fs/cgroup
tmpfs           577M     0  577M   0% /run/user/0
tmpfs           577M     0  577M   0% /run/user/1000
```

```
cat /proc/partitions
major minor  #blocks  name

   1        0       8192 ram0
   1        1       8192 ram1
   1        2       8192 ram2
   1        3       8192 ram3
   1        4       8192 ram4
   1        5       8192 ram5
   1        6       8192 ram6
   1        7       8192 ram7
   1        8       8192 ram8
   1        9       8192 ram9
   1       10       8192 ram10
   1       11       8192 ram11
   1       12       8192 ram12
   1       13       8192 ram13
   1       14       8192 ram14
   1       15       8192 ram15
   8        0       4096 sda
   8       16       4096 sdb
   8       32       8192 sdc
   8       48   62464000 sdd
   8       49       1024 sdd1
   8       50      12288 sdd2
   8       51       6144 sdd3
   8       52      12288 sdd4
   8       53     262144 sdd5
   8       54       1024 sdd6
   8       55      81920 sdd7
   8       56      16384 sdd8
   8       57      16384 sdd9
   8       58      16384 sdd10
   8       59       2048 sdd11
   8       60   62034924 sdd12
```

here is my build
https://mzh.io/asset/2018/06/64gtoendprm_ptable.img

Comment (hzhuang1) [reply l-loader/generate_ptable.sh:232]: These two partitions are required. If you don't like them, you could use your own partition instead.

Comment (mengzhuo) [reply l-loader/generate_ptable.sh:232]: ok, I think we should make system partition as big as possible, currently it only has 4GB which is apparently not enough for daily development.

**Pull request closed without merging**